src/test/java/com/fasterxml/jackson/databind/deser/enums/EnumDeserialization3638Test.java

```diff
@@ -0,0 +1,64 @@
+package com.fasterxml.jackson.databind.deser.enums;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.BaseMapTest;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class EnumDeserialization3638Test extends BaseMapTest
+{
+    /*
+    /**********************************************************
+    /* Set up
+    /**********************************************************
+     */
+
+    protected final ObjectMapper MAPPER = new ObjectMapper();
+
+    static enum Member
+    {
+        FIRST_MEMBER(0),
+        SECOND_MEMBER(1);
+
+        private int index;
+
+        private Member(int index) {
+            this.index = index;
+        }
+    }
+
+    static class SensitiveBean
+    {
+        @JsonFormat(without = JsonFormat.Feature.ACCEPT_CASE_INSENSITIVE_PROPERTIES)
+        public Member enumValue;
+    }
+
+    static class InsensitiveBean
+    {
+        @JsonFormat(with = JsonFormat.Feature.ACCEPT_CASE_INSENSITIVE_PROPERTIES)
+        public Member enumValue;
+    }
+
+    /*
+    /**********************************************************
+    /* Tests
+    /**********************************************************
+     */
+
+    public void testCaseSensitive() throws JsonProcessingException {
```

Comment:
Going forward, let's just expose `Exception` -- this only because 3.0 (`master`) removes `JsonProcessingException` (and makes Jackson exceptions unchecked) and using `Exception` removes one incompatibility wrt merging of changes.
(I'll change this no need to modify PR)